charts/aws-ebs-csi-driver/templates/node.yaml

```diff
@@ -9,6 +9,38 @@
     "node" (deepCopy $.Values.node | mustMerge $values)
   )
 }}
----
+{{- include "node" (deepCopy $ | mustMerge $args) -}}
+{{- end }}
+{{- if .Values.a1CompatibilityDaemonSet }}
+{{$args := dict
+  "NodeName" "ebs-csi-node-a1compat"
+  "Values" (dict
+    "image" (dict
+      "tag" (printf "%s-a1compat" (default (printf "v%s" .Chart.AppVersion) (.Values.image.tag | toString)))
+    )
+    "node" (dict
+      "affinity" (dict
+        "nodeAffinity" (dict
+          "requiredDuringSchedulingIgnoredDuringExecution" (dict
+            "nodeSelectorTerms" (list
+              (dict "matchExpressions" (list
+                (dict
+                  "key" "eks.amazonaws.com/compute-type"
+                  "operator" "NotIn"
+                  "values" (list "fargate")
```

Comment:
Why is this Fargate-specific check necessary?

Reply:
I'm not sure if it's absolutely necessary here, in order to make the minimal change necessary I pulled it from our defaults: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/charts/aws-ebs-csi-driver/values.yaml#L319-L327
(so the resulting change is the default + the a1 restriction, rather than the default replaced by the a1 restriction)

Even if it's not actually necessary, it definitely won't hurt, and considering we don't control what EKS fargate does it would be bad if in some future they started vending nodes with an a1 "instance type" and we tried to run on them.